datasetinsights/io/tracker/mlflow.py

```diff
@@ -0,0 +1,101 @@
+import os
+import threading
+import time
+
+import mlflow
+from google.auth.transport.requests import Request
+from google.oauth2 import id_token
+
+
+class MLFlowTracker:
+    """ MlFlow tracker class, responsible for setting host, client_id and return
+        initialized mlflow. It also refreshes the access token through daemon
+        thread.
+    Examples:
+         # Set MLTracking server UI, default is local file
+        >>> mlflow.set_tracking_uri(TRACKING_URI)
+        # New run is launched under the current experiment
+        >>> mlflow.start_run()
+        # Log a parameter (key-value pair)
+        >>> mlflow.log_param("param_name", "param_value")
+        # Log a metric (key-value pair)
+        >>> mlflow.log_metric("metric_name", "metric_val")
+        # Log an artifact (output file)
+        >>> with open("output.txt", "w") as f:
+        >>>     f.write("Hello world!")
+        >>> mlflow.log_artifact("output.txt", "run1/output/")
+        # ends the run launched under the current experiment
+        >>> mlflow.end_run()
+    Attributes:
+        REFRESH_INTERVAL: default refresh token interval
+        __mlflow: holds initialized mlflow
+    """
+
+    REFRESH_INTERVAL = (
+        3000  # every 50 minutes refresh token. Token expires in 1 hour
```

Comment:
did we check whether we can make the token last 1 week?

Reply:
I checked with Monica and Jash both. They don't have any idea about this. So I need to figure out and implement the changes. I check their setup doc and IAP PR, it does not any setting for this. I need to explore google api/IAP setup doc and implement it. 

- Is this fine I do timeout increase change after merging this code?